### ISSUE TITLE
Single PC Smooth but Dual PC with NDI Stutters / Judders?

### ISSUE THREAD

**Issue by frozenkuku** [Palakis/obs-ndi#258]
So I have been trying to find out if my dual PC setup was still not enough for a 1080p60fps stream across a stable 1gbps connection, only to find out that when I tried streaming and gaming without the NDI output (on a single PC), the stream was perfect.

Previously, I've been streaming with 2 PCs using the NDI plugin and for some reason, I keep getting stutters and judders and it hurt my eyes. Even at 720p60fps, it still had those judders. I tried and searched and tweaked many things, in fact, probably every thing. 

But when I decided to just isolate the issue without using the NDI plugin, I realized that my gaming PC whilst streaming a _veryfast_ preset, was exactly the kind of quality I was looking for. I was hoping a 2 PC setup could get me that same quality in terms of smoothness, which I can then increase the picture quality by increasing the preset to _medium_. 

Here are my specs, which I'm sure is quite an overkill. This is however, the settings when I was using the 2 PC setup, which just keeps producing judder and stutter.

**GAME PC:**

Ryzen 5 1600
GTX 1080TI OC 11gb
16gb RAM
SSD
144hz Monitor at 2560x1440 resolution
Windows 10

**STREAM PC:**

Ryzen 7 2700x
GTX 1060 6gb
16gb RAM
SSD
60hz Monitor at 1920x1080 resolution
Windows 10
(Note: Both PC's have a 650W gold PSU)

**OBS Settings for Game PC:**

Base Canvas: 2560x1440
Output: 1920x1080
Downscale filter: Lanczos
FPS: 60
Encoder: x264 (Enforce streaming encoder services)
Rate Control: CBR
Bitrate: 20000
Keyframe: 2
CPU Preset: Fast
Profile: Main

(Note: OBS usage for this PC is about 6-10%, beauty of NDI I guess)

**OBS Settings for Stream PC:**

Base Canvas: 1920x1080
Output: 1920x1080
Downscale filter: Lanczos
FPS: 60
Encoder: x264 (Enforce streaming encoder services)
Rate Control: CBR
Bitrate: 5400
Keyframe: 2
CPU Preset: Medium
Profile: Main
(Note: OBS usage for this PC is about 30-35%)

Sorry for the formatting as it is my first time posting. I hope somebody can help me...

**TL:DR**
_Streaming across NDI with a 2 PC setup produces judders and stuttering on both 720p and 1080p. However, stopping the NDI output and just streaming on 1 PC produced a super smooth stream that I want to replicate via NDI, so that I can increase the quality further by pushing my stream PC's preset._

**Comment by Helgaiden** [comment]
First, you should switch your gaming PC with your streaming PC (as in, swap the GPUs and make that secondary system your primary system).
Second, if you do that, single system streaming with that Ryzen 7 2700X is incredibly solid if a dual PC system does not become feasible for you.

My quad core i7-3770k at 4.6ghz can do 5000kbps bitrate at Medium,The R5-1600 can do it with ease as well. That 2700X is overkill and you're missing out on the gains that system offers over the 1600.

Okay now for the info.
The output settings on the gaming PC do not matter to NDI. Lets talk about the "video" section:
Base Canvas: Your monitor's resolution
Output (scaled) resolution: your desired stream resolution (though i advise 720p60 if you're streaming to twitch. I never advise 1080p60).

streaming PC:
video section- 
   base canvas: Set to the same resolution as the incoming NDI "stream." 
  Output (scaled) resolution: desired stream resolution 
  Match the filter and match the FPS setting

Output section on stream PC looks good, but make sure that "rescale output" is NOT checked. Also, you don't need "enforce streaming encoder services" checked in my experience. 

Make sure your ethernet connections are running at 1gbps speed. Hitching visible on the preview screen of the second PC USUALLY means lack of network bandwidth. If all that looks good, open a command prompt and type the following, pressing enter after each line

netsh int ip reset
netsh winsock reset

(reboot the PC)

**Comment by frozenkuku** [comment]
Thank you for your reply. 

If I am understanding you correctly, you are saying I should make my PC's like this

**Game PC:**
Ryzen 7 2700x
GTX 1080 TI

**Stream PC:**
Ryzen 5 1600
GTX 1060 6gb

The 'rescale output' was not enabled and I also unchecked the 'enforce stream encoder settings'. I have also done the command prompt resets. However, is there a way of ensuring the network bandwidth or anything related to when NDI is transmitting the video data is not compromised?

I am quite certain both PC's are using a 1.0 gbps connection with Cat 6 cable, double  / triple checked. If it does help, when I turn on task manager during the dual PC stream, the network usage is typically at about 100-130 mbps for OBS. Is that normal? Or should it be at 1.0 gbps, which doesn't make sense.

Thanks.

**Comment by Helgaiden** [comment]
Yes for the game PC/Stream Pc config stuff
Sounds like everything else is in order now. 

Go into both computers and also enable "media streaming" and see if that helps. 
https://www.isunshare.com/windows-10/turn-on-or-off-media-streaming-in-windows-10.html

**Comment by frozenkuku** [comment]
Okay. I enabled the 'media streaming'. I will now try to do a test stream using the dual PC setup. However, do note I have not changed my CPU between the 2 PC's. I will try with what you have advised so far. I feel like it may be slightly my GPU rendering some lag between 0.1% to 0.3-0.5%? 

But that's really weird for a 1080TI to have render lag when it's pushing the data through NDI to the stream PC, so technically it shouldn't be overloading. Also, GPU is usually at 40-50% on game PC with the game running and OBS.

**Comment by Helgaiden** [comment]
if you're FPS in what you're capturing is also crazy high, that can cause issues with the way OBS captures sometimes. You can right click that game capture source, click properties, and click the checkbox for "limit capture FPS" and see if it'll help

**Comment by frozenkuku** [comment]
Yes I have enabled that. 

Update: 
After doing everything, I tried streaming and it's actually worse. I'm not sure if it's the media sharing thing that is causing it, but the audio is clunky, video is spiky, way more than the initial stutter issues I had.

**Comment by Helgaiden** [comment]
Ive given that same advice to others and it has done the opposite its done to you. Something else is wrong in your configuration that i imagine might be hard to troubleshoot as a guy behind a keyboard in a different part of the world. My next bit of advice is generally this:

uninstall/remove everything NDI related from both computers. Brute force it if you want, use Revo Uninstaller Portable. Use CCleaner and do a file AND registry scan and let it clean everything up. Reboot the PC, then reinstall the stuff. This has also helped some people i've advised to do this.

**Comment by frozenkuku** [comment]
Okay. I will try that. How about if I just use Scan Converter on my game PC?

**Comment by frozenkuku** [comment]
Hey Helgaiden, sorry for a moderately long reply here. 

So I might have solved some stuff. I uninstalled all NDI related stuff and reinstalled them. Cleaned the registries. Changed game to Windowed Fullscreen Resolution. 
Changed game PC base and out resolution to 2560x1440, changed stream pc base resolution to 2560x1440 and output resolution to 1920x1080. 

NDI output shows network usage on both PCs at 190mbps to 200+ fluctuating (task manager)
I did a clean log but I played a little longer just to make sure. (hope that counts as a clean log)
Also, I removed all unnecessary scenes and cleared up the OBS on the game PC. On the stream PC, I changed all audio NDI inputs to 'audio only' and left the NDI main input at 'Highest'. Also, all bandwidth is set to 'Internal', not sure if this is supposed to be 'Network'.

Either way, from the 2 logs, running through glorp.ninja obs log analysis, there doesn't seem to be anything wrong other than the stream pc saying the preset shouldn't be lower than 'veryfast'. 

Here are the 2 logs. Game PC: https://obsproject.com/logs/ebwznLsbyFNgD3ws
Stream PC: https://obsproject.com/logs/KBg3rjG5vDFoYxHi

I also have a VOD of the 1080p stream test and 720p stream test done on an alternate account. Will be good if you can help see if it's alright now, or if there's still judder / stutter. 

1080p60fps test: https://www.twitch.tv/videos/346251892
720p60fps test: https://www.twitch.tv/videos/346248322
